### PR TITLE
/all-pending endpoint

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSBPS-707

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by mdayican — COMMENTED** [review]
do not we get pending notifications from main end point?

**Comment by apereverzin** [comment]
Just want to have list of all pending notifications without date - in case we have notifications left from previous days. Now we can only have list of notifications for a particular date.

File: src/integrationTest/java/uk/gov/hmcts/reform/notificationservice/controller/NotificationControllerTest.java [hunk 1]
```diff
@@ -367,4 +367,68 @@ void should_get_notifications_by_zip_file_name() throws Exception {
...
+        mockMvc
+                .perform(
+                        get("/notifications/all-pending")
+                )
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.count", is(2)))
+                .andExpect(jsonPath("$.notifications", hasSize(2)))
+                .andExpect(jsonPath("$.sentNotificationsCount", is(0)))
+                .andExpect(jsonPath("$.pendingNotificationsCount", is(2)))
+                .andExpect(jsonPath("$.notifications[0].id").isNotEmpty())
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/notificationservice/controller/NotificationControllerTest.java:417]: why do not we control with the real number?

File: src/integrationTest/java/uk/gov/hmcts/reform/notificationservice/controller/NotificationControllerTest.java [hunk 2]
```diff
@@ -367,4 +367,68 @@ void should_get_notifications_by_zip_file_name() throws Exception {
...
+                .andExpect(jsonPath("$.notifications[0].confirmation_id").value(notification1.confirmationId))
+                .andExpect(jsonPath("$.notifications[0].zip_file_name").value(notification1.zipFileName))
+                .andExpect(jsonPath("$.notifications[0].po_box").value(notification1.poBox))
+                .andExpect(jsonPath("$.notifications[0].container").value(notification1.container))
+                .andExpect(jsonPath("$.notifications[0].service").value(notification1.service))
+                .andExpect(jsonPath("$.notifications[0].document_control_number")
+                        .value(notification1.documentControlNumber))
+                .andExpect(jsonPath("$.notifications[1].id").isNotEmpty())
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/notificationservice/controller/NotificationControllerTest.java:425]: why do not we control with the real number?

File: src/integrationTest/java/uk/gov/hmcts/reform/notificationservice/controller/NotificationControllerTest.java (hunk 1 above)
Comment (apereverzin) [reply src/integrationTest/java/uk/gov/hmcts/reform/notificationservice/controller/NotificationControllerTest.java]: fixed

File: src/integrationTest/java/uk/gov/hmcts/reform/notificationservice/controller/NotificationControllerTest.java (hunk 2 above)
Comment (apereverzin) [reply src/integrationTest/java/uk/gov/hmcts/reform/notificationservice/controller/NotificationControllerTest.java]: fixed

**Review by mdayican — APPROVED** [review]

**Pull request merged**